### PR TITLE
CB-584 Create minimal test run that is part of deployment pipeline

### PR DESCRIPTION
**Pull request by aszegedi**
@attilapalfi92 please review. If it is Ok for you please merge.

**Note:**
Additional changes are going to be applied after finalizing the followings:
[TestContext cannot be initialized, because of 'java.lang.ClassCastException'](https://jira.cloudera.com/browse/CB-701)
[Test suite YAML Parameters section for Suite and Tests levels](https://jira.cloudera.com/browse/CB-697)

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java [hunk 1]
```diff
@@ -17,10 +19,18 @@
     @Inject
     private TestParameter testParameter;
 
+    @Inject
+    @Qualifier("cloudProviderProxy")
+    private CloudProvider cloudProvider;
```
Comment (attilapalfi92) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java]: this dependency is unnecessary.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java [hunk 2]
```diff
@@ -52,4 +62,9 @@ public Integer gatewayPort(StackV4EntityBase stackEntity) {
         }
         return Integer.parseInt(gatewayPort);
     }
+
+    @Override
+    public StackV4EntityBase stack(StackV4EntityBase stack) {
```
Comment (attilapalfi92) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java]: do not implement this method in this class.

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java [hunk 3]
```diff
@@ -17,10 +19,18 @@
...
 
+    @Inject
+    @Qualifier("cloudProviderProxy")
+    private CloudProvider cloudProvider;
+
     protected TestParameter getTestParameter() {
         return testParameter;
     }
 
+    protected CloudProvider getCloudProvider() {
```
Comment (attilapalfi92) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/newway/cloud/v2/AbstractCloudProvider.java]: this can be deleted as well.

**Comment by aszegedi** [comment]
@attilapalfi92 thanks for your help on this, I've tidied AbstractCloudProvider up.

**Pull request merged**